library/src/main/java/org/apache/apex/malhar/lib/fs/s3/S3CompactionOperator.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.apex.malhar.lib.fs.s3;
+
+import java.io.IOException;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.apache.apex.malhar.lib.fs.GenericFileOutputOperator;
+import org.apache.hadoop.fs.Path;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.DefaultOutputPort;
+
+/**
+ * This operator writes incoming tuples to HDFS files. These files will be then
+ * uploaded to S3 using S3Reconciler operator.
+ *
+ * S3TupleOutputModule uses this operator for compacting incoming tuples into
+ * files.
+ *
+ * @param <INPUT>
+ *          Type for incoming tuples. Converter needs to be defined which
+ *          converts these tuples to byte[]. Default converters for String,
+ *          byte[] tuples are provided in S3TupleOutputModule.
+ */
+@org.apache.hadoop.classification.InterfaceStability.Evolving
+public class S3CompactionOperator<INPUT> extends GenericFileOutputOperator<INPUT>
```

Comment:
I would suggest rename this class. Because, the functionality of this class is write tuples to HDFS.